src/browser/service/storage.rs

```diff
@@ -12,33 +12,75 @@ pub type Storage = web_sys::Storage;
 
 #[allow(clippy::module_name_repetitions)]
 pub fn get_storage() -> Option<Storage> {
-    web_sys::window()
-        .expect("get `window`")
-        .local_storage()
-        .ok()
-        .flatten()
+    web_sys::window()?.local_storage().ok().flatten()
 }
 
-/// Create a new store, from a serializable data structure.
-pub fn store_data<T>(storage: &Storage, name: &str, data: &T)
+/// Clear all data in local storage
+pub fn clear(storage: &Storage) {
+    let _ = storage.clear();
+}
+
+/// All of the keys in local storage
+pub fn keys(storage: &Storage) -> Vec<String> {
+    let mut keys = vec![];
+    let length = storage.length().unwrap_or(0);
```

Comment:
does it make sense to continue when you can't get storage length? - Isn't `unwrap_or(0)` another silent fail? Let's fail fast. Or if it does make sense, we can use also `.unwrap_or_default()`.